app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -30,6 +29,10 @@ class Hawkular::MiddlewareManager < ManageIQ::Providers::MiddlewareManager
 
     attr_accessor :client
 
+    def middleware_server_groups
```

Comment:
so we tend to define these as Model relations, it this case you would need to define:

```
has_many :middleware_domains, :foreign_key => :ems_id, :dependent => :destroy # you already have this one
has_many :middleware_server_groups, :through => :middleware_domains # this one you need to add, it generates the correct join then

```